### ISSUE TITLE
Critical issues with the Bitfinex Brokerage module: Faulty cash balance and errors on history access.

### ISSUE THREAD

**Issue by lucasbrynte** [QuantConnect/Lean#6578]
#### Expected Behavior
1. Properly synchronized assets / cash balance on start of algorithm.
2. When uncommenting line 27 `self.SetWarmUp(1)` in `main.py`, no error is expected.
3. When uncommenting not only line 27, but also line 38 `self.AddCrypto(self.ticker, Resolution.Hour)`, still no error is expected.

#### Actual Behavior
1. Assets / cash balance is always identically zero, although this does not correspond to the actual account balance. This can be observed from the printout of line 50 in `main.py`: `self.Log("{} - CashBook: {}".format(self.UtcTime, self.Portfolio.CashBook))`
2. When uncommenting line 27, the following error is raised: `System.Collections.Generic.KeyNotFoundException error: "'BTCUSD' wasn't found in the QuoteBars object, likely because there was no-data at this moment in time and it wasn't possible to fillforward historical data."`
3. When uncommenting lines 27 & 38 together, another error is raised, and the algorithm is terminated.

#### Reproducing the Problem
Using a Bitfinex margin account.

Local live deployment with Lean CLI:
`lean live deploy <project_name> --environment live-bitfinex --verbose`

The attached files are appended with the `.txt` extention to circumvent GitHub restrictions.

Dummy algorithm file: [main.py.txt](https://github.com/QuantConnect/Lean/files/9418530/main.py.txt)
Uncomment line 27 `self.SetWarmUp(1)`, possibly in combination with line 38 `self.AddCrypto(self.ticker, Resolution.Hour)`, in order to reproduce the behavior.

Default Lean CLI configuration (from a fresh `lean init`): [lean.json.txt](https://github.com/QuantConnect/Lean/files/9418528/lean.json.txt)
The actual configuration used differs from the above only in that the following fields are filled in with the proper credentials:
- `bitfinex-api-secret`
- `bitfinex-api-key`
- `bitfinex-organization`

#### System Information
OS: Manjaro Linux. Linux kernel: 5.4.210-1-MANJARO
Docker image: `quantconnect/lean:14490`, which as far as I understand should correspond to the HEAD of the master branch, i.e. commit `475f295ce0a5346a48d39d5ffa3f3c5015d00679`.
Lean CLI version: `1.0.103`

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

**Comment by Martin-Molinero** [comment]
Hey @lucasbrynte!

Thank you for the report. Fixes for issue 2 & 3 have been ship are in production.
Looking at issue 1 I can't seem to reproduce it, take a look at the following:

Using Margin Account Type
```python
class BasicTemplateAlgorithm(QCAlgorithm):
    def Initialize(self):
        self.SetAccountCurrency("TESTUSD")
        self.SetBenchmark(lambda x: 1)
        self.SetBrokerageModel(BrokerageName.Bitfinex, AccountType.Margin)
        self.Quit("The end")
```

![image](https://user-images.githubusercontent.com/18473240/186966583-b3a452d3-5e3d-4ad7-80d1-6891b92b79dc.png)


Using Cash Account Type
```python
class BasicTemplateAlgorithm(QCAlgorithm):
    def Initialize(self):
        self.SetAccountCurrency("TESTUSD")
        self.SetBenchmark(lambda x: 1)
        self.SetBrokerageModel(BrokerageName.Bitfinex, AccountType.Cash)
        self.Quit("The end")
```

![image](https://user-images.githubusercontent.com/18473240/186966976-ad7e20dc-aad9-4500-ad16-8080436ddaaf.png)




- **Could you please re test using the latest available versions, please make sure to pull the new lean image**
- **Review API key permissions, for example:**
![image](https://user-images.githubusercontent.com/18473240/186967180-318306cd-00c4-4e1f-add1-83eb194cbeff.png)

**Comment by lucasbrynte** [comment]
Thank you very much @Martin-Molinero for resolving the issues! This issue can be closed now.

Regarding "issue" 1, it was simply due to a mixup of accounts. Sorry for the issue report in that regard. I have used different live environment sections in `lean.json`, with different API keys configured. I believe at one point all of these got simultaneously overwritten by a set of keys provided when running Lean CLI in interactive mode (every instance of these credentials seems to have been overwritten, even inside each specific environment).

Given the fixes so far I think I'm getting close to be able to deploy a live trading algorithm again! I do however now face some other issue with the warm up period not behaving as expected... I've posted it here:
https://github.com/QuantConnect/Lean/issues/6592
